poutyne/framework/model.py

```diff
@@ -323,32 +334,77 @@ def fit_generator(self,
 
         for train_step_iterator, valid_step_iterator in epoch_iterator:
             with self._set_training_mode(True):
+                examples_in_step = 0
+
                 for step, (x, y) in train_step_iterator:
-                    step.loss, step.metrics, _ = self._fit_batch(x, y, callback=callback_list, step=step.number)
+                    zero_all_gradients = ((step.number - 1) % batches_per_step == 0)
+                    do_backprop = (step.number % batches_per_step == 0) or (step.number == steps_per_epoch)
+
                     step.size = self._get_batch_size(x, y)
 
+                    if zero_all_gradients:
+                        examples_in_step = 0
+
+                    examples_in_step += step.size
+
+                    step.loss, step.metrics, _ = self._fit_batch(x,
+                                                                 y,
+                                                                 callback=callback_list,
+                                                                 step=step.number,
+                                                                 zero_all_gradients=zero_all_gradients,
+                                                                 do_backprop=do_backprop,
+                                                                 batch_size=step.size,
+                                                                 examples_in_step=examples_in_step)
```

Comment:
I think that batch_size and examples_in_step should be None when batches_per_step is 1 so that the gradient and loss are not modified for nothing in that case.